src/odb/src/db/dbBlock.cpp

```diff
@@ -1048,6 +1057,28 @@ dbIStream& operator>>(dbIStream& stream, _dbBlock& block)
   return stream;
 }
 
+void _dbBlock::clearSystemBlockagesAndObstructions()
+{
+  dbSet<dbBlockage> blockages(this, _blockage_tbl);
+  dbSet<dbObstruction> obstructions(this, _obstruction_tbl);
+
+  for (dbBlockage* blockage : blockages) {
+    if (!blockage->isSystemReserved()) {
+      continue;
+    }
+    blockage->setIsSystemReserved(false);
+    dbBlockage::destroy(blockage);
+  }
```

Comment:
It is not safe to iterate blockages while destroying them.  The usual idiom is handled by dbNet as:
```
  static dbSet<dbNet>::iterator destroy(dbSet<dbNet>::iterator& itr);
```
You need to add a similar methods to dbBlockage and dbObstruction.

Reply:
Okay yeah. I wasn't sure if this was a thing or not. Good to know. I can just append them to a temp vector then destroy them.